File: Exercises/Easy/HappyNumber.java
package Exercises.Easy;

import java.util.HashSet;

/**
 * HappyNumber
 */
public class HappyNumber {
    public static void main(String[] args) {
        final int input = 19;
        boolean result = isHappy(input);
        System.err.println(String.format("\nIs %d a Happy Number?: %b\n", input, result));
    }

    public static boolean isHappy(int n) {
        HashSet<Integer> seenNumbers = new HashSet<Integer>();
        while (n != 1) {
            seenNumbers.add(n);
            int sumOfSquares = sumOfSquares(n);

            if (seenNumbers.contains(sumOfSquares)) {
                return false;
            } else {
                n = sumOfSquares;
            }
        }
        return true;
    }

    public static int sumOfSquares(int n) {
        int sum = 0;
        int currentDigit = 0;

        // Getting digits
        while (n > 0) {
            currentDigit = n % 10;
            sum += currentDigit * currentDigit;
            n = n / 10;
        }
        return sum;
    };
}

File: Exercises/Easy/JavaScript/HappyNumber.js
/**
 * @param {number} n
 * @return {boolean}
 */
var isHappy = function (n) {
    let seenNumbers = new Set();
    while (n > 1) {
        seenNumbers.add(n);
        let sumSquares = sumOfSquares(n);

        if (seenNumbers.has(sumSquares)) {
            return false;
        }
        n = sumSquares;
    }
    return true;
};

/**
 * @param {number} n
 * @return {number}
 */
var sumOfSquares = function (n) {
    var sum = 0;
    var currentDigit = 0;

    while (n > 0) {
        currentDigit = n % 10;
        sum += currentDigit * currentDigit;
        n = Number.parseInt(n / 10);
    }
    return sum;
};

const input = 19;

console.time("Execution time");
console.log(`Is ${input} a happy number?: ${isHappy(input)}`);
console.timeEnd("Execution time");